sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -531,4 +536,29 @@ static UUID randomUuid(long msb, long lsb) {
         // For environments using Reactor's BlockHound this will raise an exception if called in non-blocking threads.
         return new UUID(msb, lsb);
     }
+
+    /**
+     * Calls {@link Future#get(long, TimeUnit)} and returns the value if the {@code future} completes before the timeout
+     * is triggered. If the timeout is triggered, the {@code future} is {@link Future#cancel(boolean) cancelled}
+     * interrupting the execution of the task that the {@link Future} represented.
+     *
+     * @param <T> The type of value returned by the {@code future}.
+     * @param future The {@link Future} to get the value from.
+     * @param timeout The timeout value.
+     * @param unit The {@link TimeUnit} of the timeout value.
+     * @return The value from the {@code future}.
+     * @throws CancellationException If the computation was cancelled.
+     * @throws ExecutionException If the computation threw an exception.
+     * @throws InterruptedException If the current thread was interrupted while waiting.
+     * @throws TimeoutException If the wait timed out.
+     */
+    public static <T> T getFutureWithCancellation(Future<T> future, long timeout, TimeUnit unit)
```

Comment:
Should we use `Duration` instead of `long` and `TimeUnit` as params? It makes it convenient for calling libraries to just pass the duration  from the client methods.